lib/avromatic/io/union_datum.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Avromatic
+  module IO
+    class UnionDatum
+      attr_reader :member_index, :datum
+
+      def initialize(member_index, datum)
+        @member_index = member_index
+        @datum = datum
+      end
+
+      def ==(other)
+        other.is_a?(Avromatic::IO::UnionDatum) &&
+          member_index == other.member_index &&
+          datum == other.datum
+      end
+      alias_method :eql?, :==
+
+      def hash
```

Comment:
Where is this used?

Reply:
It's not currently used but it makes this class more useful as a true value object. I guess Joshua Bloch's Effective Java book burned it into my head the you should implement the hash code method whenever you implement equals :) I'll add  a unit test to cover the method.